### PR TITLE
[NoSquash] Add App Privacy Policy

### PR DESCRIPTION
**Pull request by rubenwardy**
Fixes https://github.com/minetest/minetest/issues/12809

Not sure the best way to structure this

## Preview

image

### CODE REVIEW

**Comment by rubenwardy** [comment]
Wondering if this would maybe be a better way to display the information

image

**Comment by rubenwardy** [comment]
Looks a bit terrible on a phone, which is the main place it'll be read

image

**Comment by SmallJoker** [comment]
That kinda sucks. Well then, a normal listing should serve its purpose as well. Also mentioning this because I remember having it as a TODO for the next release.

**Comment by rubenwardy** [comment]
Given the app is no longer available, this is no longer needed for 5.7

**Comment by x2048** [comment]
My 2 cents from my prior experience and reading examples on the Internet:
Privacy policy does not need to detail out every interaction between user and the services, but it is is usually written on behalf of a legal entity, like a public agreement or service agreement. Since minetest.net, server list, content DB and individual servers are managed by different legal entities, it would make sense to have a privacy policy for each of the parties. In this case privacy policy for Minetest as an application would reference the privacy policies of each party in the relevant context. The formulation could be "'Play Online' tab uses the Minetest Server List by default. See here (link) for the Privacy Policy of Minetest Server List."

Alternative to this would be to incorporate a legal entity for Minetest and transfer ownership of the individual hosted services (server list, content db, minetest.net etc.)

Disclaimer: this is not a legal advice, just sharing experience.

**Comment by rubenwardy** [comment]
> My 2 cents from my prior experience and reading examples on the Internet:
Privacy policy does not need to detail out every interaction between user and the services, but it is is usually written on behalf of a legal entity, like a public agreement or service agreement. Since minetest.net, server list, content DB and individual servers are managed by different legal entities, it would make sense to have a privacy policy for each of the parties. In this case privacy policy for Minetest as an application would reference the privacy policies of each party in the relevant context. The formulation could be "'Play Online' tab uses the Minetest Server List by default. See here (link) for the Privacy Policy of Minetest Server List."

The problem with this is that Google will review the link we share and will reject the app if it doesn't contain the information it is looking for

**Comment by rubenwardy** [comment]
Updated. Build is failing due 404s as the PR hasn't been merged yet

File: app-privacy-policy.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+
+
+## Version checking
+
+### Information collected
+
+When you open Minetest, it may contact www.minetest.net to fetch information
+about the most recent version. This HTTP request will contain the following
+information:
```
Comment (sfan5) [inline app-privacy-policy.md:56]:
```suggestion
about the most recent version. The following information will be transferred or included:
```

File: app-privacy-policy.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+* IP address
+* Minetest version
+* Platform, OS, device, etc
+
+### How it is used
+
+The version information is hosted on GitHub pages, we do not log requests.
```
Comment (sfan5) [inline app-privacy-policy.md:64]:
```suggestion
The version information is hosted on GitHub pages, which does not log requests.
```

File: app-privacy-policy.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+
+## Loading server list
+
+### Information collected
+
+When you open the Play Online tab, Minetest will request the server list. HTTP
+requests are logged with the following information:
```
Comment (sfan5) [inline app-privacy-policy.md:72]:
```suggestion
When you open the Play Online tab, Minetest will request the server list.
The following information will be transferred or included:
```

File: app-privacy-policy.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+* IP address
+* Minetest version
+* Platform, OS, device, etc
+
+### How it is used
+
+Logged HTTP requests are used to aid with debugging, development, and user
+support. They are also used to generate aggregated user statistics, see
+<https://kitsunemimi.pw/tmp/>.
```
Comment (sfan5) [inline app-privacy-policy.md:82]:
```suggestion
HTTP requests are logged to aid with debugging, development and user
support.
They may also be used to generate anonymized, aggregated user statistics.
```

File: app-privacy-policy.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+### Who has access
+
+sfan5 runs the server list and has access to the logs. The logs may be shared
+with others to aid in debugging, but care will be taken to remove any personal
+information.
+
+Aggregated statistics are shared publicly.
```
Comment (sfan5) [inline app-privacy-policy.md:90]: the intention here is good but it's not like we legally guarantee that any aggregated stats ever created are public

File: app-privacy-policy.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+### Period of retention
+
+Logged HTTP requests are stored for 6 months.
```
Comment (sfan5) [inline app-privacy-policy.md:94]: let's say 3 or less

File: app-privacy-policy.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+
+### Location
+
+The server list is currently located in the Netherlands.
+By using this service, you give permission for the data to be moved as needed.
```
Comment (sfan5) [inline app-privacy-policy.md:99]: the data the server list displays is a different matter but we're still talking about http logs with client requests, right?
there is no need ever to move the *logs* to anywhere else

File: app-privacy-policy.md (hunk 2 above)
Comment (rubenwardy) [reply app-privacy-policy.md:64]: Do we know this?

File: app-privacy-policy.md (hunk 6 above)
Comment (rubenwardy) [reply app-privacy-policy.md:94]:
```suggestion
Logged HTTP requests are stored for 3 months.
```

File: app-privacy-policy.md (hunk 7 above)
Comment (rubenwardy) [reply app-privacy-policy.md:98]: You may change the country the server is hosted in

File: app-privacy-policy.md (hunk 5 above)
Comment (rubenwardy) [reply app-privacy-policy.md:89]: Yeah, I removed this line from elsewhere should remove this one too

File: app-privacy-policy.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+### How it is used
+
+Logged HTTP requests are used to aid with debugging, development, and user
+support. They are also used to generate aggregated user statistics, see
+<https://kitsunemimi.pw/tmp/>.
+
+### Who has access
+
+sfan5 runs the server list and has access to the logs. The logs may be shared
+with others to aid in debugging, but care will be taken to remove any personal
```
[the remark this answers is not in the document]
Comment (rubenwardy) [reply app-privacy-policy.md:86]: Yeah that's the key, PII should be removed. It's a clumsy sentence though

File: app-privacy-policy.md (hunk 2 above)
Comment (sfan5) [reply app-privacy-policy.md:63]: Could always change it to "The website is hosted on GitHub pages, please refer to their privacy policy."

Comment (rubenwardy) [reply app-privacy-policy.md:63]: > When a GitHub Pages site is visited, the visitor's IP address is logged and stored for security purposes, regardless of whether the visitor has signed into GitHub or not. For more information about GitHub's security practices, see [GitHub Privacy Statement](https://docs.github.com/en/site-policy/privacy-policies/github-privacy-statement).

**Comment by celeron55** [comment]
Looks fine to me. It seems to be accurate enough. It definitely gives a good general idea.

I'm really not worried about the exact content of the privacy policy. It's bound to have inaccuracies over time as it's such a non integral non functional part of the thing it's describing.

File: app-privacy-policy.md [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+Minetest allows you to play online on multiplayer game servers. Please note that
+game servers are third-party, Minetest acts like a web browser connecting to a
+website. Therefore, please refer to the game server's privacy policy.
+
+When connecting to a game server, the following information is sent:
+
+* IP address
+* Minetest version
+* Window size, display, and input information
+* Anything you do in-game (including chat messages)
```
Comment (sfan5) [inline app-privacy-policy.md:169]: could remove this section, it's likely to become out of date and we're shifting this responsibility anyway.

**Review by sfan5 — APPROVED** [review]

File: app-privacy-policy.md (hunk 9 above)
Comment (rubenwardy) [reply app-privacy-policy.md:169]: As, we should remove the exact information and just mention to check the game server's privacy policy? I'm fine with that

**Pull request merged**